### DISCUSSION TITLE
Multiple Cameras

### DISCUSSION THREAD

**Discussion by mburge** [aaronwmorris/indi-allsky#1209] [Ideas]
It would probably take a lot of refactoring to do at the code level but have you thought about supporting capture from multiple cameras?  Capturing and storing the images would be the base. A dream would be incorporating the imagery into a unified view.

**Comment by aaronwmorris** [comment]
I flirted with the idea of managing multiple camera with indi-allsky, but abandoned the idea to just focus on stability of a single camera.

I made some effort in having multiple cameras... the database and web interface both support multiple cameras, but this idea is best realized with the SyncAPI.  It would be possible to run two cameras on two indi-allsky systems, and have one of them SyncAPI their images over to the other.  You could then view the images on a single web interface.

Many people use the SyncAPI to move images and videos to an external system on the Internet to be viewed publicly.  I sync both of my cameras to my VPS here:  https://allsky.aarmor.net/

**Reply by mburge** [reply]
Nice. I'm going to look into this.

**Comment by wjschmitt** [comment]
To achieve the multiple cameras, could indi-allsky be made to easily select from multiple indi servers running on separate machines?  Right now, I have to specify a single indi server.  I can change this, but have to restart indi-allsky.

**Reply by aaronwmorris** [reply]
You would still only be able to manage a single camera at a time.  Internally, indi-allsky only has the capability of managing a single camera at any time, this is not easily changed.

If you do not want to change indiservers, you can look at chaining multiple indiservers.  Meaning it is possible to daisy chain multiple indiservers together...

https://www.indilib.org/support/tutorials/159-indi-on-multiple-devices.html

... but perhaps I do not fully understand what you want to do.

**Reply by wjschmitt** [reply]
I didn't necessarily want to have multiple cameras at the same time.  Just to be able to switch between cameras more easily.  Maybe allow multiple indi server addresses to be added, and then to select the different cameras from the dropdown.  This is more for convenience than anything else.  I will have a look at daisy chaining.  Also, I keep forgetting that I run indi-allsky on a VM.  I can just clone it and point that one to the other indi server.  Anyway, thanks for your great program.  It keeps surprising me with what it can do.
